Sources/BrowserServicesKit/ContentBlocking/UserScripts/contentblockerrules.js

```diff
@@ -170,19 +170,21 @@
         })
     }
 
-    if (webkit.messageHandlers.processRule) {
+    if ((webkit || window.webkit).messageHandlers.processRule) {
         install()
     }
 
     function install () {
         function sendMessage (url, resourceType) {
             if (url) {
-                webkit.messageHandlers.processRule.postMessage({
-                    url: url,
-                    resourceType: resourceType === undefined ? null : resourceType,
-                    blocked: !unprotectedDomain && !isTrackerAllowlisted(topLevelUrl, url),
-                    pageUrl: topLevelUrl.href
-                })
+                try {
+                    (webkit || window.webkit).messageHandlers.processRule.postMessage({
```

Comment:
Perhaps the try catch is the only difference here.

As mentioned perhaps there's an execution context that doesn't have the message handlers available.